front-end/src/Views/ProjectPage.js

```diff
@@ -147,6 +153,19 @@ export default class HomePage extends Component {
         this.setState({ username: name, userlastname: lastname });
         });
   };
+
+  getProfileNameById = (userId) => {
+    axios.get(`${config.API_URL}/api/profiles/${userId}/`, { headers: { 'Content-Type': 'Application/json'} })
+      .then(res => {
+        var name = res.data.name;
+        var mname = res.data.middle_name;
+        var lastname = res.data.last_name;
+        const val = name + " " + mname + " " + lastname;
+        console.log(val);
+        return val;
+      });

```

Comment:
I think, you should use getProject function to get the profile's name via this.state.member, because in api call, return will not work

Reply:
Thanks, now I have realized that I cannot return the value from that api call; but my problem is that: I want to get the names of profiles which have sent collaboration requests, so I cannot get the profile name via this.state.member because it returns the name of the owner of that project; however I want to get the names of request senders. In uniqueIds in line 351, I am holding an array which consists of the ids of the profiles that sent colab requests and I want to get the names from these ids. Any other suggestions for it? 
Thanks